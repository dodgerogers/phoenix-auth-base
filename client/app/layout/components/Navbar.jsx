import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Menu, Button, Header, Container } from 'semantic-ui-react';
import CurrentUser from '../../hocs/CurrentUser';
import LoginModal from '../../authentication/components/LoginModal';

const Navbar = (props) => {
  const loggedInLinks = () => (
    <Menu.Item>Logged in as {props.currentUser.get('name')}</Menu.Item>
  );

  const nonLoggedInLinks = () => (
    <Menu.Item>
      <LoginModal
        trigger={<Button color="teal">Login</Button>}
        {...props}
      />
    </Menu.Item>
  )

  return (
    <navbar>
      <Menu attached='top'>
        <Container>
          <Menu.Item>
            <Header>
              Teebox.io
            </Header>
          </Menu.Item>
          <Menu.Menu position='right'>
            {props.currentUser ? loggedInLinks() : nonLoggedInLinks()}
          </Menu.Menu>
        </Container>
      </Menu>
    </navbar>
  );
}

Navbar.propTypes = {
  currentUser: ImmutablePropTypes.map,
};

export { Navbar as PureComponent };
export default CurrentUser(Navbar);
