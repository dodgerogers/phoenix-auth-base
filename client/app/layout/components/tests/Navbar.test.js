import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import Navbar, { PureComponent } from '../Navbar';


describe('Navbar', () => {
  const pureComponent = (props) => shallow(
    <PureComponent {...props} />
  );

  it('matches pure component snapshot', () => {
    const wrapper = pureComponent({});

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('matches pure component when currentUser is present snapshot', () => {
    const props = { currentUser: fromJS({ name: 'name' }) };
    const wrapper = pureComponent(props);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
