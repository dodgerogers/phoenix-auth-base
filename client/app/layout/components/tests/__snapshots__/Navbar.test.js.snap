// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches pure component snapshot 1`] = `
<navbar>
  <Menu
    fixed="top"
    style={
      Object {
        "boxShadow": "none",
      }
    }
  >
    <Container>
      <MenuItem>
        <Header>
          Teebox.io
        </Header>
      </MenuItem>
      <MenuMenu
        position="right"
      >
        <Connect(ModalTrigger)
          id="SIGN_IN_MODAL"
          text="Login"
          wrapper={[Function]}
        />
        <Connect(ModalTrigger)
          id="REGISTRATION_MODAL"
          text="Sign up"
          wrapper={[Function]}
        />
      </MenuMenu>
    </Container>
  </Menu>
</navbar>
`;

exports[`Navbar matches pure component when currentUser is present snapshot 1`] = `
<navbar>
  <Menu
    fixed="top"
    style={
      Object {
        "boxShadow": "none",
      }
    }
  >
    <Container>
      <MenuItem>
        <Header>
          Teebox.io
        </Header>
      </MenuItem>
      <MenuMenu
        position="right"
      >
        <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={true}
          deburr={false}
          icon="dropdown"
          item={true}
          minCharacters={1}
          noResultsMessage="No results found."
          openOnFocus={true}
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          trigger={
            <UserAvatar
              user={
                Immutable.Map {
                  name: "Billy",
                }
              }
            />
          }
          wrapSelection={true}
        >
          <DropdownMenu>
            <DropdownHeader>
              My Account
            </DropdownHeader>
            <DropdownItem>
              <a
                onClick={[Function]}
              >
                Sign out
              </a>
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </MenuMenu>
    </Container>
  </Menu>
</navbar>
`;
